src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@AutoValue
+public abstract class JavaDocComment {
+  public enum COMMENT_TYPE {
+    COMMENT,
+    HTML_P,
+    HTML_UL,
+    HTML_OL,
+    SAMPLE_CODE,
+  }
+
+  public static List<COMMENT_TYPE> commentList = new ArrayList<>();;
+
+  public abstract Optional<String> deprecated();
```

Comment:
Would prefer to stick with @Nullable over Optional. Since we will "know" that some values can never be null, running into null pointer exceptions will help us detect bugs early on.

Reply:
got it, change the optional fields to `nullable` annotation.